### ISSUE TITLE
JIT trace does not work correctly with pyro.plate

### ISSUE THREAD

**Issue by fehiepsi** [uber/pyro#1685]
I am not sure if this is a bug or I am using "pyro.plate" in a wrong way.

The following code shows the issue. I got this issue while trying to train a HMM model. Things work well for non-jit version. If we call `torch.autograd.grad(loss_jit(x), (x,))` two times, it will throw RuntimeError: "Can not go backward 2 times for the graph".

```
import pyro
import pyro.distributions as dist
import pyro.poutine as poutine
import torch

def model():
    with pyro.plate("plate", 2):
        pyro.sample("x", dist.Normal(0, 1))
    #pyro.sample("x", dist.Normal(0, 1).expand((2,)).to_event())  # this works!

trace = poutine.trace(model).get_trace()

def loss(x):
    trace.nodes["x"]["value"] = x
    return poutine.trace(poutine.replay(model, trace=trace)).get_trace().log_prob_sum()

x = torch.tensor([0., 1.], requires_grad=True)
loss_jit = torch.jit.trace(loss, x, check_trace=False)
print(loss_jit(x))  # -2.3379
print(loss_jit(x))  # -4.6758
```

**Comment by fehiepsi** [comment]
Excellent! Thanks a lot @eb8680 for a clear explanation! Things work smoothly now. :)

**Comment by fritzo** [comment]
Nice sleuthing, @eb8680! We should fix that line in `Trace.log_prob_sum()` to use `result = result + _` rather than `+=`.

**Comment by fehiepsi** [comment]
Closed via #1687.
